feat: add domain as an openfeature concept (#229)
open-feature/spec: specification/sections/01-flag-evaluation.md

OpenFeature.getProviderMetadata();
```

> It must be possible to access provider metadata using a `domain`.

Reviewer: ```suggestion
It's possible to access provider metadata using a `domain`.
```

If a provider has not be registered under the requested domain, the default provider metadata is returned.

```typescript
// example provider accessor

> OpenFeature.getProviderMetadata("domain-1");

Reviewer 2: We might need to check SDK implementations when this spec get merged. Java already expose this [^1] and I created a follow-up for Go SDK [^2]. Needs to check other SDKs 

[^1]: https://github.com/open-feature/java-sdk/blob/main/src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java#L53-L54
[^2]: https://github.com/open-feature/go-sdk/issues/247

```

See [provider](./02-providers.md), [domain](../glossary.md#domain) for details.

### Creating clients

#### Requirement 1.1.6

> The `API` **MUST** provide a function for creating a `client` which accepts the following options:
>
> - domain (optional): A logical string identifier for binding clients to provider.

```java
// example client creation and retrieval
OpenFeature.getClient();
```

> It must be possible to create a client that is associated with a `domain`.

Reviewer: ```suggestion
It's possible to create a client that is associated with a `domain`.
```

```suggestion
It's possible to create a client that is associated with a `domain`.
```

I try to avoid must/should/etc (RFC2119 language) in non-normative sections. I recommend speaking as if things are already implemented in these sections (like the sentence below: "the client will..."

OpenFeature.getProviderMetadata();
```

> TODO: Add text explaining that this is for a namespace

Author: Action the TODO
